state-sync/src/request_manager.rs

```diff
@@ -493,58 +477,56 @@ impl RequestManager {
             self.update_score(peer, PeerScoreUpdateType::TimeOut);
         }
 
-        // increment multicast level if this request is also multicast-timed-out
+        // Increase the multicast network level if this request has also hit a multicast timeout
         let multicast_start_time = self.get_multicast_start_time(version).unwrap_or(UNIX_EPOCH);
         if is_timeout(multicast_start_time, self.multicast_timeout) {
-            let new_multicast_level = self.multicast_level.checked_add(1).ok_or_else(|| {
-                Error::IntegerOverflow("New multicast level has overflown!".into())
-            })?;
-            let max_multicast_level = self
-                .upstream_config
-                .upstream_count()
-                .checked_sub(1)
-                .ok_or_else(|| Error::IntegerOverflow("Upstream count has overflown!".into()))?; // multicast_level (=network preference) is 0-indexed
-            let new_multicast_level = std::cmp::min(new_multicast_level, max_multicast_level);
-            self.update_multicast(new_multicast_level, Some(version));
+            // Move to the next multicast network level
+            let new_multicast_network_level = match self.multicast_network_level {
+                NetworkId::Validator => NetworkId::vfn_network(),
+                _ => NetworkId::Public,
+            };
+            self.update_multicast_network_level(new_multicast_network_level, Some(version));
         }
         Ok(timeout)
     }
 
+    /// The validator network and vfn network are always considered upstream when state syncing, as
+    /// multicasting will prioritize sending chunk requests to the most important network first.
+    /// If the peer is on the public network, we only consider it upstream (i.e., a valid peer
+    /// to send chunk requests to) if we connected to it.
     fn is_upstream_peer(&self, peer: &PeerNetworkId, origin: ConnectionOrigin) -> bool {
-        let is_network_upstream = self
-            .upstream_config
-            .get_upstream_preference(peer.raw_network_id())
-            .is_some();
-        // check for case whether the peer is a public downstream peer, even if the public network is upstream
-        if is_network_upstream && peer.raw_network_id() == NetworkId::Public {
-            origin == ConnectionOrigin::Outbound
-        } else {
-            is_network_upstream
-        }
+        peer.raw_network_id().is_validator_network()
+            || peer.raw_network_id().is_vfn_network()
```

Comment:
As discussed offline, this method is only really used in the case of failover to other networks, i.e., it checks when given a peer notification that it can send state sync requests to the peer. I agree though, the name is not great. To avoid the name overloading, I think we can name it `is_valid_state_sync_peer`. Is that clearer to you?

Reply:
I think you can leave it for now, we just need to probably later take a step back and figure out how to untangle these overloaded terms.